### ISSUE TITLE
Mangafox isn’t working anymore and AMR stopped adding new mangas.

### ISSUE THREAD

**Issue by lcangelica** [AllMangasReader-dev/AMR#157]
Similar to the error happening to mangahere described in issue #156, mangafox isn't working as well.
AMR stopped adding new mangas to my list and the ones already on it doesn't update when I read new chapters.
Every time I open chrome AMR shows all my list as new, 600+ mangas showing as recently updated.

**Comment by Stubbs9410** [comment]
same here pls help

**Comment by Snipeon** [comment]
As said in #156, @twluo would be looking into it when he has time. As for now, I'm brushing up my JS so I can hopefully find a fix.

**Comment by monkey8113** [comment]
Its a free plugin, i don't hope for much. Heck, even if its paid plugin. I'll pay to use it! This plugin has change my manga life and stayed with me many years. I'd be very happy if its fixed but for now we can wait for a fix by anyone that willing to spend their free time fixing it. This plugin stayed alive by contributors good will. Its sad that i don't know how to code.

**Comment by twluo** [comment]
@Stubbs9410 @Snipeon @monkey8113 @lcangelica 

I'm not sure what exactly is wrong with MangaFox since I don't use it very much. But try out my new change and tell me if there's anything else broken. I fixed the issue of updating and adding mangas also I think it should fixe the 600+ mangas showing as recently updated. I didn't have too much time to test. But you can test for me :)
https://github.com/twluo/mirrors/commit/9460086d9b70643bcc13a513d930ff5c2ae3ceac
https://github.com/twluo/AMR/commit/24f5be23f0e9563a2bee0e847dccb6294dc94eb0

**Comment by Snipeon** [comment]
@twluo good man, you fixed it all before I could take a crack at it! Nice! 

Who knew adding a simple http: into the equation would work...

**Comment by king37sniper** [comment]
It's able to download ?

**Comment by Snipeon** [comment]
Yes. Though I would say it's Mangahere and Mangafox having trouble with their servers now.

**Comment by king37sniper** [comment]
From here ? http://allmangasreader.com

**Comment by Snipeon** [comment]
Grab the repo from @twluo and manually load it, @braiam hasn't approved of his changes yet...

**Comment by king37sniper** [comment]
OK

**Comment by Snipeon** [comment]
@twluo Mangafox just turned back on https, it's broken again.

**Comment by twluo** [comment]
I made some changes so that it recognizes https://mangafox.me but it seems that its getting an error when calling mangafox so not sure it works sometimes it doesn't work other times.

**Comment by Snipeon** [comment]
both Mangahere and Mangafox is now sometimes responding and other times not. I feel that is the problem.

**Comment by Snipeon** [comment]
Hmm... it seems that Mangafox and Mangahere have both disallowed even loading several chapter lists at once... 

The chapter view is also pretty unstable, sometimes loading and sometimes not...

**Comment by twluo** [comment]
Where you got this information from?

**Comment by Snipeon** [comment]
bug testing? I don't know what other reason for there to be a 503 error where it would otherwise load normally other than instability or the server side disallowing it.

I tried mass loading the chapter lists in my AMR, but I just received a bunch of 503s. But when I tried to load them one by one, all of em were 200s. So yea, it's either it's disallowed or the server is too unstable to handle a bunch of requests at once. I'm leaning more on the former.

**Comment by DakuTree** [comment]
It's most likely this is some kind of bot protection like what bato.to added about a year ago.

The only solution I've found for this is to grab the chapter/images off the mobile reader, as that doesn't appear to have the bot protection.  
This doesn't work for series licensed in the US however, as those aren't visible on the mobile reader (even if you're outside the US and can view them on the normal site).

**Comment by twluo** [comment]
Is there a workaround for mangahere?

**Comment by king37sniper** [comment]
We neeeeeeeeeeeeeeeeeeeed u @braiam 
Pls Help :(

**Comment by Snipeon** [comment]
I'll just quote @braiam here:
> This is a problem of MangaFox that understands that people loading several images at once are abusing (I have no other way to interpret a 503 error where the image loads fine some times). The only solution is to talk with MF so they allow it, or just disable the chapter view.

He was talking about the problem when chapter view was broken on Mangafox. And seeing how Mangafox and Mangahere are owned by the same idiot who insists on making others' life miserable just to further his own profits(go check it out), it would just be an uphill battle. 

On that note, I'll say that you did a really good job, @twluo. If you really did want to fix everything, I'll say that it would just increase the loading times. By a lot.

**Comment by king37sniper** [comment]
It's fine, i can wait a little bit longer when loading manga
As long as i still can track my mangas list and watch it in all page mode

**Comment by Snipeon** [comment]
@king37sniper then in that case @twluo's version can do that just fine now... Just enable this in AMR...
![This](https://user-images.githubusercontent.com/15312503/31051524-3ebef1b6-a69d-11e7-8437-de367e73864a.png)

**Comment by twluo** [comment]
MangaHere doesn't load chapters all at once T_T

**Comment by king37sniper** [comment]
Nope, in my case it still broken

**Comment by Snipeon** [comment]
@twluo well yep, there's a 503 preventing the script from running correctly...

@king37sniper have you updated your manga lists? Meaning export your list, add https, import it back in...

**Comment by king37sniper** [comment]
Yep, i did all of it

**Comment by monkey8113** [comment]
Actually i don't think that's the problem, as Manga Viewer plugin still function perfectly.I just don't like how it store and notify new chapter.

**Comment by drehren** [comment]
If you have problems loading all pages, you can check the "Load chapter pages in order" setting, that will only request one page at a time... that fixed MangaFox for me when it still worked.

**Comment by noodle27** [comment]
I'm having problems loading all pages on Mangahere right now (I applied @twluo's quick fix and it fixed Mangahere, things worked nicely but yesterday it happens again). Already checked "Load chapter pages in order" and "Save banwidth" but it doesn't help.

**Comment by Snipeon** [comment]
@monkey8113 good idea, I'll cross reference MV and AMR, See how they differ in loading chapters and stuff. Though be warned that it could take a couple of months, I need to brush up my JS and shit first.

**Comment by twluo** [comment]
MV also has a limit on how many mangas you can store which is one of the reasons I'm not a fan. Is anyone else having issues when you restart your computer the mangalist is not updated to the latest point? Its regressing to a previous points.

**Comment by twluo** [comment]
Is there a reason why you guys stick to mangafox or mangahere? Is it just because you started with it and migrating will be a pain in the ass? I'm thinking of writing a script to migrate all my mangas to mangareader. Is anyone interested on it? (if I manage to finish it)

**Comment by Snipeon** [comment]
@twluo I use all three, on top of batoto. It's just that there are some manga which do not exist on mangareader and vice versa. And mangareader takes quite a bit to update the newest chapters, on top of regular mess ups of the newest chapter formatting.

**Comment by noodle27** [comment]
@twluo I don't use mangafox since they compressed the image like an asshole, but on mangahere it's still at an acceptable size without a huge arse watermark on the page, and they do have some series that are banned on batoto (like goblin slayer for example) so I'm still using it as back up (for series that are not available on batoto)

**Comment by monkey8113** [comment]
I used mangafox because most of the manga available else where will be there. While some are not available in mangahere. But both of them lose to chinese manga website which AMR stop supporting long time ago. The chinese version of all manga is normally faster and more complete.

**Comment by coderpsy** [comment]
@Snipeon isnt batoto have same issue. it doesnt track latest chapter.(at least for me)

Is it possible issue can be solved with this ?
https://stackoverflow.com/questions/8476233/get-past-request-limit-in-crawling-a-web-site.

When I change my ip and import mangas batoto works for a day or two then all my latest chapters become null(cant find,cant open link but if manga have a new chapter I get notification :) ). i think website somehow tracks crawlers and ip ban users.

**Comment by monkey8113** [comment]
@twluo after some testing with your AMR fix, and exporting my data where i replace the http to https it all works well. But i do realise AMR tooks forever to load manga now, this problem seem to happens before the https switch. When i test it on MV however, it loads normally. No idea what they do differently over there but it works. 
To be exact, it happens when i enable "Load scans in the order of the chapter". When its off however it loads a lot faster but quite a few page would fail to load and would need to click on "Try to reload" button to reload it.
I cant do much to help, but i can tell you my findings just incase it might be helpful to you.
Thanks a lot for the fix! Really appreciate it!

**Comment by king37sniper** [comment]
@braiam 
Maybe he's on vacation, so sad =))

**Comment by king37sniper** [comment]
Can u guys add new manga ?
In my case, arm alway show "No chapters found yet..."

**Comment by noodle27** [comment]
I can add them normally. Did you try make AMR reload the chapter list?

**Comment by king37sniper** [comment]
Yes, mangahere and fox broken, i can't add new manga
Mangareader work fine

**Comment by Snipeon** [comment]
No problem here as well.

**Comment by king37sniper** [comment]
:(

**Comment by CrisisJam** [comment]
Hi, I'm not sure if it is just me but I check Mangafox every day and it seems that no new mangas or updates have been added since Aug 8th. Could this be checked, please?

**Comment by king37sniper** [comment]
> Hi, I'm not sure if it is just me but I check Mangafox every day and it seems that no new mangas or updates have been added since Aug 8th. Could this be checked, please?

Mangafox  has already changed to fanfox,net.
https://discord.gg/YGBxwZ and a new AMR ext released, check it out.
